### ISSUE TITLE
Region annotation shows as normal annotations

### ISSUE THREAD

**Issue by salehabbas** [grafana/grafana#36283]
**What happened**:
I tried to create a **region annotation** (_regions event_) via _Annotations HTTP API_ but when checking the panel, I see it as a normal annotation.

**What you expected to happen**:
I expected to see _a region annotation_ with **_start_** and **_end_** time on the panel/dashboard.

**How to reproduce it (as minimally and precisely as possible)**:

- Create a region annotation via Annotations HTTP API
```
curl -H "Content-Type: application/json" --request POST --data '{"time":1624965740790, "timeEnd": 1624965825058, "tags":["test"], "text":"something started and finished in that interval"}' http://localhost:3000/api/annotations
```
- Verify that the annotation was successfully created
```
curl http://localhost:3000/api/annotations?tags=test
[
  {
    "id": 9,
    "alertId": 0,
    "alertName": "",
    "dashboardId": 0,
    "panelId": 0,
    "userId": 0,
    "newState": "",
    "prevState": "",
    "created": 1624968431459,
    "updated": 1624968431459,
    "time": 1624965740790,
    "timeEnd": 1624965825058,
    "text": "something started and finished in that interval",
    "tags": [
      "test"
    ],
    "login": "admin",
    "email": "admin@localhost",
    "avatarUrl": "/avatar/46d229b033af06a191ff2267bca9ae56",
    "data": {}
  }
]
```

- Checked the dashboard and cannot see the region annotation (it's rendered as normal annotation)

So basically did not get that results: https://grafana.com/static/img/docs/annotations/region-annotation.gif
Instead got: https://grafana.com/static/img/docs/annotations/annotation.gif

**Anything else we need to know?**:

**Environment**:
- Grafana version: v8.0.3 (cae5c5e46b) 
- Data source type & version: Prometheus / 2.28.0
- OS Grafana is installed on: Alpine Docker image
- User OS & Browser:  MacOs Catalina 10.15.7 / Google Chrome 91.0.4472.77
- Grafana plugins: N/A
- Others: N/A

**Comment by dsotirakis** [comment]
Hello @salehabbas , thanks for raising the issue. Talked with the team, better support for annotations will be out in Grafana 8.1, including the issue you mentioned. Thanks for understanding :)

**Comment by dprokop** [comment]
Related work: https://github.com/grafana/grafana/pull/36220
